### PR TITLE
DM-21343: query with ORDER BY and SELECT * fails

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: core/modules/qana/TablePlugin.cc [hunk 1]
```diff
@@ -215,7 +215,11 @@ TablePlugin::applyLogical(query::SelectStmt& stmt,
 
     if (stmt.hasOrderBy()) {
         matchTableRefs(context, stmt.getOrderBy(), false);
-        matchValueExprs(context, stmt.getOrderBy(), true);
+        // The columns in the ORDER BY statement do not have to have a match in the select list, yet.
+        // This allows a query like `SELECT * ... ORDER BY col`, where col exists in the table (and so this
+        // is a valid query) but is not yet explicitly named. The existance of the ORDER BY columns will be
```
Comment (andy-slac) [inline core/modules/qana/TablePlugin.cc]: https://en.wiktionary.org/wiki/existance

File: core/modules/query/ValueExpr.h [hunk 2]
```diff
@@ -91,8 +91,28 @@ class ValueExpr {
...
+     * The state of the member variable `aliasIsUserDefined` will not be changed.
+     */
     void setAlias(std::string const& alias);
+
+    /**
+     * @brief Set the Alias of the ValueExpr, and indicate if the alias is user defined or not.
+     *
+     * The state of the member variable `aliasIsUserDefined` will be changed.
+     */
+    void setAlias(std::string const& alias, bool aliasIsUserDefined);
```
Comment (andy-slac) [inline core/modules/query/ValueExpr.h]: I'm not sure how these two overloads of `setAlias` work w.r.t. `aliasIsUserDefined`. I think it's clear what happens if you call second overload - the state becomes either "user-defined" or "not-user-defined". But if you call first overload the state is "no-one-gives-a-damn"? Is that state useful? It's likely just my confusion as I do not know much about this code and I generally dislike modifiable state. 🙂

File: core/modules/query/ValueExpr.h [hunk 3]
```diff
@@ -233,10 +261,12 @@ class ValueExpr {
     // determine if this object is the same as or a less complete description of the passed in object.
     bool isSubsetOf(ValueExpr const& valueExpr) const;
 
+    void setToAliasOnly();
```
Comment (andy-slac) [inline core/modules/query/ValueExpr.h]: Comment would be useful here.

File: core/modules/qana/TablePlugin.cc (hunk 1 above)
Comment (n8pease) [reply core/modules/qana/TablePlugin.cc]: brutal! 😆

Comment (n8pease) [reply core/modules/qana/TablePlugin.cc]: (oh, as noted above)

**Comment by fritzm** [comment]
Closing this pending further rework (see comments in Jira issue).

**Pull request closed without merging**